include/deal.II/base/work_stream.h

```diff
@@ -663,6 +663,163 @@ namespace WorkStream
+                .emplace([it = i,
+                          idx,
+                          &data,
+                          &sample_scratch_data,
+                          &sample_copy_data,
+                          &copy_datas,
+                          &worker]() {
+                  ScratchData *scratch_data = nullptr;
+
+                  ScratchDataList &scratch_data_list = data.get();
+                  // See if there is an unused object. if so,
+                  // grab it and mark it as used.
+                  for (auto &p : scratch_data_list)
+                    {
+                      if (p.currently_in_use == false)
+                        {
+                          scratch_data       = p.scratch_data.get();
+                          p.currently_in_use = true;
+                          break;
+                        }
+                    }
+                  // If no element in the list was found, create
+                  // one and mark it as used.
+                  if (scratch_data == nullptr)
+                    {
+                      scratch_data_list.emplace_back(
+                        std::make_unique<ScratchData>(sample_scratch_data),
+                        true);
+                      scratch_data =
+                        scratch_data_list.back().scratch_data.get();
+                    }
+
+                  // Create a unique copy data object where this
+                  // worker's work will be stored.
+                  auto &copy = copy_datas[idx];
+                  copy       = std::make_unique<CopyData>(sample_copy_data);
+                  worker(it, *scratch_data, *copy.get());
+
+                  // Find our currently used scratch data and
+                  // mark it as unused.
+                  for (auto &p : scratch_data_list)
+                    {
+                      if (p.scratch_data.get() == scratch_data)
+                        {
+                          Assert(p.currently_in_use == true,
+                                 ExcInternalError());
+                          p.currently_in_use = false;
+                        }
+                    }
+                })
+                .name("worker");
+
+            // Create a copier task.
+            tf::Task copier_task = taskflow
+                                     .emplace([idx, &copy_datas, &copier]() {
+                                       copier(*copy_datas[idx].get());
+                                       copy_datas[idx].reset();
+                                     })
+                                     .name("copy");
+
+            // Ensure the copy task runs after the worker task.
+            worker_task.precede(copier_task);
+
+            // Ensure that only one copy task can run at a time.
+            if (!last_copier.empty())
+              last_copier.precede(copier_task);
+            last_copier = copier_task;
+          }
+
+        executor.run(taskflow).wait();
```

Comment:
I think it would be useful to start the implementation with something like "First build a graph of tasks that need to be run" and then here say "Now actually let the graph execute".

Reply:
I meant for these to be in-code comments, just to be clear.